src/k2/module/k23si/Module.cpp

```diff
@@ -887,28 +895,38 @@ K23SIPartitionModule::handleWrite(dto::K23SIWriteRequest&& request, FastDeadline
     }
 
     // check to see if we should push or is this a write from same txn
-    if (versions.size() > 0 && versions[0].status == dto::DataRecord::WriteIntent) {
-        auto& rec = versions[0];
-        auto& rqmtr = request.mtr;
-
-        if (rec.txnId.mtr != rqmtr) {
-            // this is a write request finding a WI from a different transaction. Do a push with the remaining
-            // deadline time.
-            K2LOG_D(log::skvsvr, "Partition: {}, different WI found for key {}", _partition, request.key);
-            return _doPush(request.collectionName, rec.key, rec.txnId, request.mtr, deadline)
-                .then([this, request = std::move(request), deadline](auto&& retryChallenger) mutable {
-                    if (retryChallenger) {
-                        K2LOG_D(log::skvsvr, "Partition: {}, write push retry for key {}", _partition, request.key);
-                        return handleWrite(std::move(request), deadline);
-                    }
-                    // challenger must fail
-                    K2LOG_D(log::skvsvr, "Partition: {}, write push challenger lost for key {}", _partition, request.key);
-                    return RPCResponse(dto::K23SIStatus::AbortConflict("incumbent txn won in write push"), dto::K23SIWriteResponse{});
-                });
-        }
+    if (versions.WI.has_value() && versions.WI->txnId.mtr != request.mtr) {
```

Comment:
I think this should only be done for IPR WI. If the WI is aborted, we should probably remove it from memory